### ISSUE TITLE
Why is there a difference in loss value logged using self.log ?

### ISSUE THREAD

**Issue by sbp-dev** [PyTorchLightning/pytorch-lightning#5247]
#### What is your question?
(New to Lightning) The progress bar shows loss for each step. Before logging other metrics using self.log, wanted to sanity test it to log the loss again with a different name.

**Please check the last 2 lines of the code**. I'm logging the same loss value under "train_loss" that is being returned so I expected to see the "loss" and "train_loss_step" values to be the same in the progress bar but they aren't. The values don't seem to be time-shifted.

Which value should I interpret as the loss of the current mini-batch and why is there a difference?
#### Code
```
class LitCNN(pl.LightningModule):
    def __init__(self):
        super(LitCNN, self).__init__()
        # Model preparation
        model = torchvision.models.resnet34(pretrained=True)
        # Freeze layers
        for params in model.parameters():
            params.requires_grad = False
        # Change the final fully-connected layer to match our number of classes
        model.fc = nn.Linear(in_features=512, out_features=5, bias=True)
        self.model = model
        
        self.metric_accuracy = pl.metrics.Accuracy()
        self.loss = nn.CrossEntropyLoss()

    def forward(self, x):
        # called with self(x)
        return self.model(x)#.argmax(1) # Return class ID

    def training_step(self, batch, batch_nb):
        # REQUIRED
        X, y = batch
        logits = self.model(X)
        loss = self.loss(logits, y)
        
        self.log('train_loss', loss, on_step=True, on_epoch=True, prog_bar=True)
        return {'loss': loss}
```
   
![PL_loss2](https://user-images.githubusercontent.com/54526452/103014172-5892e480-4564-11eb-88c2-aa7bf4e19866.jpg)
![PL_loss1](https://user-images.githubusercontent.com/54526452/103014174-59c41180-4564-11eb-8b92-6243e5b301b6.jpg)

#### What have you tried?
I've experimented with the "reduce_fx" arg in self.log but it seems to get invoked only on epoch end. 
#### What's your environment?

 - OS: Win 10
 - Packaging: Conda
 - Version: 1.1.0

**Comment by awaelchli** [comment]
The loss value is averaged over a moving window [1] for easier reading, the other values I think are not.

> Which value should I interpret as the loss of the current mini-batch
The train_loss_step or the value you can read from the tensorboard graph (you can set the log interval to 1 if you want)

[1] https://pytorch-lightning.readthedocs.io/en/stable/new-project.html#logging

**Comment by sbp-dev** [comment]
Thanks for the reference, makes sense now! Could you please point me to the code where this moving average is computed? 
I like to see the loss of each batch as is and logging a new one clutters the p-bar so I guess I'll implement a custom progress bar using callbacks. Couldn't find detailed examples of implementing one though, any pointers on that?
Thanks again!

**Comment by awaelchli** [comment]
Here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/b4d926baf25d1c02e3fdedafee85526b8ee1ea97/pytorch_lightning/trainer/supporters.py#L24

I wouldn't waste the time to make a custom  progress bar just for this. Inspecting variables through the progress bar is just a pain, it's not going to help you much. I suggest you inspect the per-batch loss values in the logger, you can set `log_every_n_steps=1` to get a value every step.

**Comment by sbp-dev** [comment]
Thanks for the prompt help. I discovered the Tensorboard logger soon after and life is indeed easier with that!
